freqtrade/pairlist/IPairList.py

```diff
@@ -90,6 +91,16 @@ def _validate_pair(self, ticker) -> bool:
         """
         raise NotImplementedError()
 
+    def gen_pairlist(self, cached_pairlist: List[str], tickers: Dict) -> List[str]:
+        """
+        Generate the pairlist
+        :param cached_pairlist: Previously generated pairlist (cached)
+        :param tickers: Tickers (from exchange.get_tickers()).
+        :return: List of pairs
+        """
```

Comment:
I think this docstring should mention that
* this method is only called for the first ~pairlist~ handler in the list
* it shall not be implemented for filters

Also - i would document these changes to `developer.md` - it's describing the (current) implementation of a new pairlist quite well  (we should do that for other things too) - but we need to be carefull that it's kept uptodate (at least for the most part).

Reply:
> this method is only called for the first pairlist handler in the list

hah! should it be called if StaticPairList is in the middle of the chain, rewriting all previous results with the contents of the whitelist from the config? (that's current (previous) behavior)

Ideally, this usage should raise the exception, but I do not know yet how to achieve it with this (proposed here) implementation...

Reply:
just check `pairlist_pos` in the constructor of the ~pairlist~ pairhandler - if it must be first (which is the case for staticPairlist) - then raise if it's not the first ... ? 

This way, control is given to the handler... :)

Reply:
Yeah, but my intention was to get rid of this position completely in the future...

Reply:
i wouldn't do this - also for future Pairlists i can think of it being useful to know if you're the first or last pairlist ...

Reply:
well, it's not what I was going to make in the nearest future anyway, so let's see if this attribute is useful...

so, the question is what should be the expected behavior for StaticPairList when it's placed somewhere in the middle of the chain: rewrite the pairlist prepared by the previous handlers in the chain (with a call to gen_pairlist()), do nothing (with a call to filter_pairlist(), which simply returns pairlist, prepared by the previous handlers in the chain) or raise OperationalException, noting that this Pairlist Handler should only be used at the first position in the chain)?

Reply:
Exception - StaticPairlist has no other functionality - and overwriting everything others did previously would be ... highly unexpected (but that's how it's currently i think).